contracts/src/v0.8/automation/test/v2_3/BaseTest.t.sol

```diff
@@ -356,40 +355,44 @@ contract BaseTest is Test {
     );
   }
 
-  function _transmit(uint256 id, Registry registry) internal {
+  function _transmit(uint256 id, Registry registry, bytes4 selector) internal {
     uint256[] memory ids = new uint256[](1);
     ids[0] = id;
-    _transmit(ids, registry);
+    _transmit(ids, registry, selector);
   }
 
-  function _transmit(uint256[] memory ids, Registry registry) internal {
-    uint256[] memory upkeepIds = new uint256[](ids.length);
-    uint256[] memory gasLimits = new uint256[](ids.length);
-    bytes[] memory performDatas = new bytes[](ids.length);
-    bytes[] memory triggers = new bytes[](ids.length);
-    for (uint256 i = 0; i < ids.length; i++) {
-      upkeepIds[i] = ids[i];
-      gasLimits[i] = registry.getUpkeep(ids[i]).performGas;
-      performDatas[i] = new bytes(0);
-      uint8 triggerType = registry.getTriggerType(ids[i]);
-      if (triggerType == 0) {
-        triggers[i] = _encodeConditionalTrigger(
-          AutoBase.ConditionalTrigger(uint32(block.number - 1), blockhash(block.number - 1))
-        );
-      } else {
-        revert("not implemented");
+  function _transmit(uint256[] memory ids, Registry registry, bytes4 selector) internal {
```

Comment:
It was a bit confusing for me why we're passing `bytes4(0)` everywhere in the tests. I would suggest doing something like this...

```solidity
// internal functions, these can be used by AutomationRegistry.t.sol

// tests single upkeep, expects success
function _transmit(uint256 id, Registry registry) internal {
  uint256[] memory ids = new uint256[](1);
  ids[0] = id;
  _handleTransmit(ids, registry, bytes4(0));
}

// tests multiple upkeeps, expects success
function _transmit(uint256[] memory ids, Registry registry) internal {
  _handleTransmit(ids, registry, bytes4(0));
}

// tests single upkeep, expects revert
function _transmitAndExpectRevert(uint256 id, Registry registry, bytes4 selector) internal {
  uint256[] memory ids = new uint256[](1);
  ids[0] = id;
  _handleTransmit(ids, registry, selector);
} 

// private function not exposed to actual testing contract

function _handleTransmit(uint256[] memory ids, Registry registry, bytes4 selector) private {
  // same logic from the current _transmit() function
}

```

Reply:
done